internal/framework6provider/identity_resource_test.go

```diff
@@ -32,25 +32,50 @@ func TestIdentityResource(t *testing.T) {
 				ConfigStateChecks: []statecheck.StateCheck{
 					statecheck.ExpectIdentity("framework_identity.test", map[string]knownvalue.Check{
 						"id":   knownvalue.StringExact("id-123"),
-						"name": knownvalue.StringExact("my name is john"),
+						"name": knownvalue.StringExact("john"),
 					}),
+					statecheck.ExpectKnownValue("framework_identity.test", tfjsonpath.New("id"), knownvalue.StringExact("id-123")),
+					statecheck.ExpectKnownValue("framework_identity.test", tfjsonpath.New("name"), knownvalue.StringExact("john")),
 				},
 			},
+			// Typically you don't need to test all of these different import methods,
+			// but this just a smoke test for passing state + identity data through.
+			{
+				ImportState:     true,
+				ResourceName:    "framework_identity.test",
+				ImportStateKind: resource.ImportCommandWithID,
+			},
+			{
+				ImportState:     true,
+				ResourceName:    "framework_identity.test",
+				ImportStateKind: resource.ImportBlockWithID,
+			},
+			{
+				ImportState:     true,
+				ResourceName:    "framework_identity.test",
+				ImportStateKind: resource.ImportBlockWithResourceIdentity,
+			},
+		},
+	})
+}
+
+func TestIdentityResource_identity_changes(t *testing.T) {
+	resource.UnitTest(t, resource.TestCase{
+		TerraformVersionChecks: []tfversion.TerraformVersionCheck{
+			tfversion.SkipBelow(tfversion.Version1_12_0),
+		},
+		ProtoV6ProviderFactories: map[string]func() (tfprotov6.ProviderServer, error){
+			"framework": providerserver.NewProtocol6WithError(New()),
+		},
+		Steps: []resource.TestStep{
 			{
 				Config: `resource "framework_identity" "test" {
 					name = "jerry"
 				}`,
-				ConfigPlanChecks: resource.ConfigPlanChecks{
-					PreApply: []plancheck.PlanCheck{
-						plancheck.ExpectResourceAction("framework_identity.test", plancheck.ResourceActionUpdate),
-					},
-				},
-				ConfigStateChecks: []statecheck.StateCheck{
-					statecheck.ExpectIdentity("framework_identity.test", map[string]knownvalue.Check{
-						"id":   knownvalue.StringExact("id-123"),
-						"name": knownvalue.StringExact("my name is john"), // doesn't get updated, since identity should not change.
-					}),
-				},
+				// The resource is hardcoded to refresh with the same identity, based off the name attribute during create.
+				// Resources are currently not allowed to change identities at any time, so core will return an error message
+				// after the post-apply refresh.
+				ExpectError: regexp.MustCompile(`Error: Provider produced different identity`),
```

Comment:
This test will need to be commented out for now, since Core recently removed this validation and we haven't re-introduced it in Framework / SDKv2

Reply:
I updated the go mod to use the changes introduced in this PR: https://github.com/hashicorp/terraform-plugin-framework/pull/1137